### PR TITLE
Add FXIOS-9955 [Microsurvey] Border requirement with new toolbar

### PR DESCRIPTION
**Pull request by cyndichin**
## :scroll: Tickets
[Jira ticket](https://mozilla-hub.atlassian.net/browse/FXIOS-9955)
[Github issue](https://github.com/mozilla-mobile/firefox-ios/issues/21865)

## :bulb: Description

**Requirements**
Add border requirement for microsurvey with the new toolbar
- When survey is shown, hide border in between survey and toolbar
- When survey is shown and search bar is at bottom, hide border between search and survey
- When survey is dismissed, show border as normal

**Summary**
* Added a new `.none` case for `AddressToolbarBorderPosition` to be explicit that there is no border instead of using `nil`
* Utilized existing `isBottomSearchBar` instead of `isToolbarPositionBottom` because it seems that the state was not updated in time and the existing field provided a more reliable check after changing the address bar position

There are now three cases where we want to update the border:
1. **Microsurvey is shown** - update borders depending if toolbar position is top or bottom
2. **Scrolling** - updated to only change the address toolbar when on top since that seems to only be the concerned when scrolling
3. **Toolbar position has changed** - updated to check if microsurvey is shown, if so then we update the border for both nav and address bar

## :pencil: Checklist
You have to check all boxes before merging
- [x] Filled in the above information (tickets numbers and description of your work)
- [x] Updated the PR name to follow our [PR naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [ ] Wrote unit tests and/or ensured the tests suite is passing
- [ ] When working on UI, I checked and implemented accessibility (minimum Dynamic Text and VoiceOver)
- [ ] If needed, I updated documentation / comments for complex code and public methods
- [ ] If needed, added a backport comment (example `@Mergifyio backport release/v120`)

## Screenshots
See iPhone, iPad should not be impacted since the main issue is the bottom toolbars

https://github.com/user-attachments/assets/c832163c-fd95-47ff-b85e-1e9d04ffd683

### CODE REVIEW

**Comment by cyndichin** [comment]
@thatswinnie @MattLichtenstein I believe I remember there were some Toolbar State tests, but I am unable to find them for some reason. Could you point me to them or am I recalling incorrectly?

File: firefox-ios/Client/Frontend/Browser/Toolbars/Redux/ToolbarMiddleware.swift [hunk 1]
```diff
@@ -222,6 +222,7 @@ final class ToolbarMiddleware: FeatureFlaggable {
     }
 
     // MARK: - Border
+    // For the top placement of the address bar, the border is only visible on scroll
```
Comment (cyndichin) [inline firefox-ios/Client/Frontend/Browser/Toolbars/Redux/ToolbarMiddleware.swift:225]: do you know the decision behind this, would it be helpful to add a comment on why we are doing this? For user experience?

**Comment by cyndichin** [comment]
> Nit: I think it would be great to have some comments for when a border should be shown or hidden with Microsurvey.

thanks @thatswinnie !~ I added some more comments, let me know if they are helpful or if you have any suggestions! I had one question on the scrolling requirement so I can better flesh out the `why` for that requirement.

File: firefox-ios/Client/Frontend/Browser/Toolbars/Redux/ToolbarMiddleware.swift (hunk 1 above)
Comment (thatswinnie) [reply firefox-ios/Client/Frontend/Browser/Toolbars/Redux/ToolbarMiddleware.swift:225]: In Figma it's written that it's a design choice to only show the border on scroll for the top address toolbar placement.

**Comment by thatswinnie** [comment]
> > Nit: I think it would be great to have some comments for when a border should be shown or hidden with Microsurvey.
> 
> thanks @thatswinnie !~ I added some more comments, let me know if they are helpful or if you have any suggestions! I had one question on the scrolling requirement so I can better flesh out the `why` for that requirement.

Thanks @cyndichin ! Looks great! And I commented on your question. Hope that helps!

File: firefox-ios/Client/Frontend/Browser/Toolbars/Redux/ToolbarMiddleware.swift (hunk 1 above)
Comment (cyndichin) [reply firefox-ios/Client/Frontend/Browser/Toolbars/Redux/ToolbarMiddleware.swift]: thank you!

**Review by MattLichtenstein — APPROVED** [review]

**Pull request merged**